server/src/main/java/com/flipkart/varadhi/web/v1/authz/AuthZHandlers.java

```diff
@@ -0,0 +1,205 @@
+                                true,
+                                Optional.empty()
+                        ),
+                        new RouteDefinition(
+                                HttpMethod.PUT,
+                                "/orgs/:org/teams/:team/policy",
+                                Set.of(hasBody, authenticated),
+                                new LinkedHashSet<>(),
+                                this.setIAMPolicyHandler(ResourceType.TEAM),
+                                true,
+                                Optional.empty()
+                        ),
+                        new RouteDefinition(
+                                HttpMethod.GET,
+                                "/projects/:project/policy",
+                                Set.of(authenticated),
+                                new LinkedHashSet<>(),
+                                this.getIAMPolicyHandler(ResourceType.PROJECT),
+                                true,
+                                Optional.empty()
+                        ),
+                        new RouteDefinition(
+                                HttpMethod.PUT,
+                                "/projects/:project/policy",
+                                Set.of(hasBody, authenticated),
+                                new LinkedHashSet<>(),
+                                this.setIAMPolicyHandler(ResourceType.PROJECT),
+                                true,
+                                Optional.empty()
+                        ),
+                        new RouteDefinition(
+                                HttpMethod.GET,
+                                "/projects/:project/topics/:topic/policy",
+                                Set.of(authenticated),
+                                new LinkedHashSet<>(),
+                                this.getIAMPolicyHandler(ResourceType.TOPIC),
+                                true,
+                                Optional.empty()
+                        ),
+                        new RouteDefinition(
+                                HttpMethod.PUT,
+                                "/projects/:project/topics/:topic/policy",
+                                Set.of(hasBody, authenticated),
+                                new LinkedHashSet<>(),
+                                this.setIAMPolicyHandler(ResourceType.TOPIC),
+                                true,
+                                Optional.empty()
+                        )
+                )
+        ).get();
+    }
+
+    @Override
+    public List<RouteDefinition> get() {
+        return Stream.of(
+                getDebugHandlers(),
+                getPolicyHandlers()
+        ).flatMap(List::stream).toList();
+    }
+
+    public Handler<RoutingContext> getIAMPolicyHandler(ResourceType resourceType) {
+        return (routingContext) -> {
+            String resourceId = routingContext.getResourceIdFromPath(resourceType);
```

Comment:
Putting getResourceIdFromPath in extension is a code smell.
This method should lie here as a normal method.

Extension methods are pure utility methods. They dont handle business logic.

Reply:
done